internal/proxy_service/tcp/proxy_service.go

```diff
@@ -0,0 +1,144 @@
+package tcp
+
+import (
+	"fmt"
+	"io"
+	"net"
+
+	validation "github.com/go-ozzo/ozzo-validation"
+
+	"github.com/cyberark/secretless-broker/internal"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+)
+
+// Zeroizes the values of the fetched credentials. We don't want to
+// rely on garbage collection for this (it might be slow and/or only free them) so
+// we manually clear
+func zeroizeCredentials(backendCredentials map[string][]byte) {
+	for _, credentialBytes := range backendCredentials {
+		for i := range credentialBytes {
+			credentialBytes[i] = 0
+		}
+	}
+}
+
+func duplexStream(source io.ReadWriter, destination io.ReadWriter) <-chan error {
+	errChan := make(chan error)
+	go func() {
+		errChan <- stream(source, destination)
+	}()
+	go func() {
+		errChan <- stream(destination, source)
+	}()
+	return errChan
+}
+
+func stream(source io.Reader, destination io.Writer) error {
+	for {
+		if _, err := io.Copy(destination, source); err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+type proxyService struct {
+	connector           tcp.Connector
+	retrieveCredentials internal.CredentialsRetriever
+	listener            net.Listener
+	done                bool
+	error               chan error
+}
+
+// NewProxyService constructs a new instance of a TCP ProxyService. The
+// constructor takes a TCP Connector, CredentialResolver and Listener.
+// A TCP ProxyService is able to Connect with Credentials then subsequently stream
+// bytes between client and target service
+func NewProxyService(
+	connector tcp.Connector,
+	resolveCredentials internal.CredentialsRetriever,
```

Comment:
Nit: switching the language here seems unnecessary.  I'd just called the local var `retrieveCredentials`